deployment/ccip/changeset/cs_rmn_curse_uncurse.go

```diff
@@ -0,0 +1,275 @@
+}
+
+type RMNCurseAction struct {
+	ChainSelector  uint64
+	SubjectToCurse Subject
+}
+
+type CurseAction func(e deployment.Environment) []RMNCurseAction
+
+type RMNCurseConfig struct {
+	MCMS         *MCMSConfig
+	CurseActions []CurseAction
+	Reason       string
+}
+
+func (c RMNCurseConfig) Validate(e deployment.Environment) error {
+	state, err := LoadOnchainState(e)
+
+	if err != nil {
+		return errors.Errorf("failed to load onchain state: %v", err)
+	}
+
+	if len(c.CurseActions) == 0 {
+		return errors.Errorf("curse actions are required")
+	}
+
+	if c.Reason == "" {
+		return errors.Errorf("reason is required")
+	}
+
+	validSelectors := e.AllChainSelectors()
+	validSubjects := make([]Subject, 0, len(validSelectors)+1)
+	for _, selector := range validSelectors {
+		validSubjects = append(validSubjects, SelectorToSubject(selector))
+	}
+	validSubjects = append(validSubjects, GlobalCurseSubject())
+
+	for _, curseAction := range c.CurseActions {
+		result := curseAction(e)
+		for _, action := range result {
+			targetChain := e.Chains[action.ChainSelector]
+			targetChainState := state.Chains[action.ChainSelector]
+
+			if err := commoncs.ValidateOwnership(e.GetContext(), c.MCMS != nil, targetChain.DeployerKey.From, targetChainState.Timelock.Address(), targetChainState.RMNRemote); err != nil {
+				return fmt.Errorf("chain %s: %w", targetChain.String(), err)
+			}
+
+			if !slices.Contains(validSelectors, action.ChainSelector) {
+				return errors.Errorf("invalid chain selector %d for chain %s", action.ChainSelector, targetChain.String())
+			}
+
+			if !slices.Contains(validSubjects, action.SubjectToCurse) {
+				return errors.Errorf("invalid subject %x for chain %s", action.SubjectToCurse, targetChain.String())
+			}
+		}
+	}
+
+	return nil
+}
+
+type Subject = [16]byte
+
+func SelectorToSubject(subject uint64) Subject {
+	var b Subject
+	binary.BigEndian.PutUint64(b[8:], subject)
+	return b
+}
+
+// CurseLaneOnlyOnSource curses a lane only on the source chain
+// This will prevent message from source to destination to be initiated
+// One noteworthy behaviour is that this means that message can be sent from destination to source but will not be executed on the source
+func CurseLaneOnlyOnSource(sourceSelector uint64, destinationSelector uint64) CurseAction {
+	// Curse from source to destination
+	return func(e deployment.Environment) []RMNCurseAction {
+		return []RMNCurseAction{
+			{
+				ChainSelector:  sourceSelector,
+				SubjectToCurse: SelectorToSubject(destinationSelector),
+			},
+		}
+	}
+}
+
+func CurseLane(sourceSelector uint64, destinationSelector uint64) CurseAction {
+	// Bidirectional curse between two chains
+	return func(e deployment.Environment) []RMNCurseAction {
+		return append(
+			CurseLaneOnlyOnSource(sourceSelector, destinationSelector)(e),
+			CurseLaneOnlyOnSource(destinationSelector, sourceSelector)(e)...,
+		)
+	}
+}
+
+func CurseChain(chainSelector uint64) CurseAction {
```

Comment:
Doc comment. Good heuristic is to doc comment all public (exported) methods and structs.